test/docker-e2e/e2e_upgrade_test.go

```diff
@@ -0,0 +1,221 @@
+package docker_e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"celestiaorg/celestia-app/test/docker-e2e/dockerchain"
+
+	signaltypes "github.com/celestiaorg/celestia-app/v5/x/signal/types"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/celestiaorg/celestia-app/v5/pkg/user"
+	tastoradockertypes "github.com/celestiaorg/tastora/framework/docker"
+	"github.com/celestiaorg/tastora/framework/testutil/wait"
+	tastoratypes "github.com/celestiaorg/tastora/framework/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// TestCelestiaAppUpgrade tests app version upgrade using the signaling mechanism.
+func (s *CelestiaTestSuite) TestCelestiaAppUpgrade() {
+	if testing.Short() {
+		s.T().Skip("skipping celestia-app major upgrade test in short mode")
+	}
+
+	tag, err := dockerchain.GetCelestiaTagStrict()
+	s.Require().NoError(err)
+
+	const (
+		MinTargetAppVersion uint64 = 3 // v2 to v3 was the first upgrade using signaling
+
+		// blocked on https://github.com/celestiaorg/celestia-app/issues/5289
+		// TODO: uncomment this once the issue is fixed
+		// MaxTargetAppVersion = appconsts.Version
```

Comment:
This line has to be uncommented to enable app version 5 to be tested and the next line should be removed. Otherwise it tests up to appver 4

Reply:
I tested that out and it did pass for me